### PR TITLE
Update shoulda to version 3.6.0

### PR DESCRIPTION
**Pull request by mjankowski**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -70,7 +70,7 @@ group :test do
   gem 'rack-test', require: 'rack/test'
   gem 'mocha', require: false
   gem 'bourne', require: false
-  gem 'shoulda', require: false
+  gem 'shoulda'
```
Comment (sonalkr132) [inline Gemfile]: Why are we removing `require: false`?

File: test/test_helper.rb [hunk 2]
```diff
@@ -49,3 +49,10 @@ class ActionDispatch::IntegrationTest
 class SystemTest < ActionDispatch::IntegrationTest
   include Capybara::DSL
 end
+
+Shoulda::Matchers.configure do |config|
```
Comment (sonalkr132) [inline test/test_helper.rb]: No changelog :( Why are we adding this?
Can you please also undo change of BUNDLED WITH?

File: Gemfile (hunk 1 above)
Comment (mjankowski) [reply Gemfile]: Updated commit msg to expand on version bump

File: test/test_helper.rb (hunk 2 above)
Comment (mjankowski) [reply test/test_helper.rb]: Updated commit to explain the config block here and undo the bundler version bump.

**Comment by sonalkr132** [comment]
Merged as 66145bd3

**Pull request closed without merging**